shimmy/meltingpot_compatibility.py

```diff
@@ -42,41 +43,28 @@ class MeltingPotCompatibilityV0(ParallelEnv, EzPickle):
 
     def __init__(
         self,
-        substrate_name: str,
+        env: meltingpot.python.utils.substrates.substrate.Substrate,
+        substrate_name: str | None = None,
         render_mode: str | None = None,
         max_cycles: int = MAX_CYCLES,
-        env: meltingpot.python.utils.substrates.substrate.Substrate | None = None,
     ):
         """Wrapper that converts a openspiel environment into a pettingzoo environment.
 
         Args:
-            substrate_name (str): name of meltingpot substrate to load
+            env (meltingpot.python.utils.substrates.substrate.Substrate): already initialized melting pot env to wrap
+            substrate_name (Optional[str]): name of melting pot substrate to load (instead of already initialized env)
             render_mode (Optional[str]): render_mode
             max_cycles (Optional[int]): maximum number of cycles (steps) before termination
-            env (Optional[meltingpot.python.utils.substrates.substrate.Substrate]): existing meltingpot env to use
 
         """
-        EzPickle.__init__(self, substrate_name, render_mode, max_cycles, env)
+        EzPickle.__init__(self, env, substrate_name, render_mode, max_cycles)
 
-        self.max_cycles = max_cycles
-        if env is not None:
-            self._env = env
+        if substrate_name is not None:
+            self._env = utils.load_substrate(substrate_name)
         else:
```

Comment:
We need to check that `env` is not None as well

Reply:
Yeah I’ll make it raise an error if you try to specify both an env and a substrate name and say which it will default to (the env I think makes most sense).